pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -320,6 +320,19 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, r.Update(context.TODO(), cd)
 	}
 
+	// Set platform label on the ClusterDeployment
+	if platform := getClusterPlatform(cd); cd.Labels[hivev1.HiveClusterPlatformLabel] != platform {
```

Comment:
This looks odd. If the cd already has a label and it does not match with `getClusterPlatform()` we should not overwrite it as this condition should not come. Which means there is bigger issue somewhere. So we need to add warning here that it does not match. We should only add the label when it does not exist.

Reply:
I disagree. Hive should be authoritative on the label. If the user removes or changes the label, Hive should change it back. Hive could add a warning that it has been changed, or Hive could log an info whenever setting it, but Hive should always ensure that the label is set properly.

Reply:
I don't expect any process would change or alter this label, but that said, my thoughts were along the lines of what @staebler said.. this controller should be authoritative for the label, and replace it with what it knows to be the correct value.

We could possibly add a warning inside this if, that if the label is not an empty string, then log out something with the old and new values as we overwrite.. but personally I'm not sure that's necessary.

Reply:
I strongly feel we should add logging to at least tell user when we do this. So it would be explicit then implicit. Sending a PR ro fix it.